src/Pos/Types/Core.hs

```diff
@@ -134,7 +133,9 @@ mkCoinPortion x
 -- Internally 'CoinPortion' stores 'Word64' which is divided by 'coinPortionDenominator'
 -- to get actual value. So some rounding may take place.
 unsafeCoinPortionFromDouble :: Double -> CoinPortion
-unsafeCoinPortionFromDouble x = assert (0 <= x && x <= 1) $ CoinPortion v
```

Comment:
@gromakovsky By the way, I traced this function back to you (you wrote it about 27 days ago).
Is there a problem if I remove the `assert` here? Is the assert here absolutely necessary?
I couldn't get the exception to be thrown because, as you later said, `assert`s do not work if optimizations are used. Moreover, all the other coin converting functions don't use it, so I found it strange.